### PR TITLE
Fix #1958. Remove release notes for old version of Docker Compose.

### LINKED ISSUE #1958: [DOC] Outdated release notes for 1.2.0 on https://docs.docker.com/compose/
I was going through the website and noticed the release notes section is outdated:
https://docs.docker.com/compose/ - v1.2.0

The latest release:
https://github.com/docker/compose/releases - v1.4.0

Should we remove the information from the website and link to the "releases" page on GitHub directly?

### PR DESCRIPTION
**Pull request by charleswhchan**
Replace by link to the latest CHANGELOG in GitHub.

Signed-off-by: Charles Chan <charleswhchan@users.noreply.github.com>

### CODE REVIEW

**Comment by dnephin** [comment]
LGTM

File: docs/index.md [hunk 1]
```diff
@@ -206,17 +206,8 @@ At this point, you have seen the basics of how Compose works.
...
-For complete information on this release, see the [1.2.0 Milestone project page](https://github.com/docker/compose/wiki/1.2.0-Milestone-Project-Page).
-In addition to bug fixes and refinements, this release adds the following:
-
-* The `extends` keyword, which adds the ability to extend services by sharing  common configurations. For details, see
-[PR #1088](https://github.com/docker/compose/pull/1088).
-
-* Better integration with Swarm. Swarm will now schedule inter-dependent
-containers on the same host. For details, see
-[PR #972](https://github.com/docker/compose/pull/972).
+To see a detail list of changes for past and current releases of Docker Compose,
```
Comment (aanand) [inline docs/index.md]: "detail" -> "detailed"

**Comment by aanand** [comment]
Thanks! Minor typo, then it's good to go.

File: docs/index.md (hunk 1 above)
Comment (charleswhchan) [inline docs/index.md]: Fixed.

**Pull request merged**